### PR TITLE
EASY-1946: zipfile is not checked for validity before parsing

### PR DESCRIPTION
**Pull request by stephanHogenboom**
Fixes EASY-1946

#### When applied it will
* return a 400 (malformed zip) if the a zipped file misses some bytes (corrupted)

- [x] add empty zipFile test

#### Related pull requests on github
repo                       | PR                | note
-------------------------- | ----------------- | ----
easy-                      | [PR#](PRlink)     | ..

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala [hunk 1]
```diff
@@ -73,16 +72,16 @@ package object servlets extends DebugEnhancedLogging {
...
+      Try(Option(zipInputStream.getNextEntry)) match {
+        case Success(Some(firstEntry: ZipEntry)) => for {
           _ <- extract(firstEntry)
           _ <- Stream
             .continually(zipInputStream.getNextEntry)
             .takeWhile(Option(_).nonEmpty)
             .map(extract)
             .failFastOr(Success(()))
         } yield ()
+        case _ => Failure(MalformedZipException(s"No entries found."))
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]:
```suggestion
        case _ => Failure(MalformedZipException(s"No entries found."))
                case Success(None) => Failure(MalformedZipException(s"No entries found."))
        case Failure(e) => Failure(e)
```

File: src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala [hunk 2]
```diff
@@ -73,16 +73,18 @@ package object servlets extends DebugEnhancedLogging {
...
+        case Success(Some(firstEntry: ZipEntry)) => for {
           _ <- extract(firstEntry)
           _ <- Stream
             .continually(zipInputStream.getNextEntry)
             .takeWhile(Option(_).nonEmpty)
             .map(extract)
             .failFastOr(Success(()))
         } yield ()
+        case Success(None) => Failure(MalformedZipException(s"No entries found."))
+        case Failure(_: EOFException) => Failure(MalformedZipException(s"No entries found."))
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.deposit/servlets/package.scala]: you can merge these two lines:

```scala
case Success(None) | Failure(_: EOFException) => Failure(MalformedZipException(s"No entries found."))
```

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala [hunk 3]
```diff
@@ -167,6 +167,40 @@ class UploadSpec extends DepositServletFixture {
...
+    val absoluteTarget = (testDir / "drafts" / "foo" / uuid.toString / "bag/data" / relativeTarget).createDirectories()
+    post(
+      uri = s"/deposit/$uuid/file/$relativeTarget",
+      params = Iterable(),
+      headers = Seq(fooBarBasicAuthHeader),
+      files = bodyParts
+    ) {
+      absoluteTarget.list.size shouldBe 0
+      status shouldBe 400
+      body shouldBe s"ZIP file is malformed. No entries found."
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala]:
```suggestion
      body shouldBe "ZIP file is malformed. No entries found."
```

File: src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala [hunk 4]
```diff
@@ -167,6 +167,40 @@ class UploadSpec extends DepositServletFixture {
...
+    val uuid = createDeposit
+    val relativeTarget = "path/to/dir"
+    val absoluteTarget = (testDir / "drafts" / "foo" / uuid.toString / "bag/data" / relativeTarget).createDirectories()
+    post(
+      uri = s"/deposit/$uuid/file/$relativeTarget",
+      params = Iterable(),
+      headers = Seq(fooBarBasicAuthHeader),
+      files = bodyParts
+    ) {
+      body shouldBe s"ZIP file is malformed. No entries found."
```
Comment (rvanheest) [inline src/test/scala/nl.knaw.dans.easy.deposit/servlets/UploadSpec.scala]:
```suggestion
      body shouldBe "ZIP file is malformed. No entries found."
```

**Pull request merged**